examples/common/generate_callback.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2022 MosaicML Examples authors
+# SPDX-License-Identifier: Apache-2.0
+
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Periodically log generations to wandb from a set of prompts."""
+from typing import List
+
+import wandb
+from composer.core import Callback, State
+from composer.loggers import Logger, WandBLogger
+from composer.utils import dist, ensure_tuple
+
+
+class Generate(Callback):
+
+    def __init__(self, prompts: List[str], **kwargs):
+        """Periodically log generations to wandb from a set of prompts.
+
+        In the main view for a run, there will be a table that will show the _last_ logged generations.
+        To compare previous iterations of the generations, you need to
+        1. Click on the run
+        2. Click on "artifacts" in the menu on the left side of the screen
+        3. Click on one of the artifacts called "predictions"
+        4. Click on the "files" tab
+        5. Click on "predictions.table.json"
+        6. On the left hand side, there are different versions of the table produced throughout training. Select one of these.
+        7. Now, when you hover over other versions, there will be a "compare" button, which will allow you to compare the currently
+            selected version to the version you add via compare.
+
+        Args:
+            prompts (List[str]): The list of prompts you would like to produce generations for
+            kwargs: All kwargs well be passed along to the call to generate. This is for things like `do_sample`, `top_p`, etc
+        """
+        self.prompts = prompts
+        self.generate_kwargs = kwargs
+
+    def init(self, state: State, logger: Logger):
+        if dist.get_global_rank() == 0:
+            for destination in ensure_tuple(logger.destinations):
+                if isinstance(destination, WandBLogger):
```

Comment:
do we care about [other loggers](https://docs.mosaicml.com/en/latest/trainer/logging.html#available-loggers)?

Reply:
or is this a problem for when the callback is upstreamed to composer?

Reply:
im gonna say no; this is a wandb specific implementation. can be adapted in the future if desired.

Reply:
basically this callback is a no-op if you're not using wandb. I think that is ok for now. if/when upstreamed i can think about supporting other loggers as makes sense.